adapter/internal/oasparser/utills/open_api_utils.go

```diff
@@ -51,25 +52,28 @@ func hasPrefix(buf []byte, prefix []byte) bool {
 	return bytes.HasPrefix(trim, prefix)
 }
 
-// FindSwaggerVersion finds the openapi version ("2" or "3") for the given
-// openAPI json content.
-func FindSwaggerVersion(jsn []byte) string {
+// FindAPIDefinitionVersion finds the openapi version ("2" or "3") or Async API version for the given
+// API Definition.
+func FindAPIDefinitionVersion(jsn []byte) string {
 	var version string = "2"
 	var result map[string]interface{}
 
 	err := json.Unmarshal(jsn, &result)
 	if err != nil {
-		logger.LoggerOasparser.Error("json unmarsheliing err when finding the swaggerVersion : ", err)
+		logger.LoggerOasparser.Error("Error while JSON unmarshalling to find the API definition version.", err)
 	}
 
 	if _, ok := result["swagger"]; ok {
-		version = "2"
+		return "2"
 	} else if _, ok := result["openapi"]; ok {
-		version = "3"
-	} else {
-		logger.LoggerOasparser.Warn("swagger file version is not defined. Default version set as to 2 ")
-		return version
-
+		return "3"
+	} else if versionNumber, ok := result["asyncapi"]; ok {
+		if strings.HasPrefix(versionNumber.(string), "2") {
+			return "asyncapi_2"
+		}
+		logger.LoggerOasparser.Errorf("AsyncAPI version %v is not supported.", versionNumber.(string))
+		return "not_supported_asyncapi_version" + versionNumber.(string)
```

Comment:
```suggestion
		return "not_supported_asyncapi_version " + versionNumber.(string)
```

Reply:
With the current behaviour it will return as:
`return errors.New("API version not specified or not supported for the version" + not_supported_asyncapi_version 3)`
in - https://github.com/Thushani-Jayasekera/product-microgateway/blob/3ddcf27d6b343cfae44a36814efaca423a651470/adapter/internal/oasparser/model/mgw_swagger.go#L1235

And changing as such will give the following compile error:
```
cannot use fmt.Errorf("not_supported_asyncapi_version: %q", versionNumber) (value of type error) as string value in return statement
```
Therefore, can we keep it in the original form

Reply:
I meant if we return (string, error) we can use this.

Reply:
Let's keep a space before versionNumber